x-pack/plugins/enterprise_search/public/applications/app_search/utils/results/index.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export const flattenField = (fieldName: string, fieldValue: object): Array<[string, object]> => {
+  const flattened: Array<[string, object]> = [];
+
+  if (typeof fieldValue === 'object' && !Object.keys(fieldValue).includes('raw')) {
+    Object.entries(fieldValue).map(([propName, value]) => {
```

Comment:
suggestion: If you're just iterating over this array to produce a wholly separate one, you just want a traditional for loop (`for (const [propName, value] of Object.entries) {...}`)

Map is designed to transform every field in an array, and using it has some performance and memory consequences. If you do want to use map to get more immutable/functional-style code, you would probably end up with some form of reduce instead.